### ISSUE TITLE
ruleset exclude subpackage not working

### ISSUE THREAD

**Issue by sajov** [squizlabs/PHP_CodeSniffer#474]
```
<rule ref="Squiz.Commenting.FileComment">
     <exclude name="Squiz.Commenting.FileComment.MissingSubpackageTag" />
     <exclude name="Squiz.Commenting.FileComment.MissingCopyrightTag" />
     <exclude name="Squiz.Commenting.FileComment.IncorrectAuthor" />
     <exclude name="Squiz.Commenting.FileComment.IncorrectCopyright" />
     <exclude name="Squiz.Commenting.FileComment.DuplicateAuthorTag" />
 </rule>
```
still get error for missing subpackege

**Comment by aik099** [comment]
What you mean by `subpackage`?

Per my knowledge:

* the `exclude` tag can't be used to disable individual errors in the included sniffs (it can be used to remove some sniffs, when you've including whole ruleset
* you can disable individual errors of included sniffs through usage of `severity`:

```xml
<rule ref="Squiz.Commenting.FileComment.MissingSubpackageTag">
    <severity>0</severity>
</rule>
```

**Comment by aik099** [comment]
Here is annotated ruleset for the reference: http://pear.php.net/manual/en/package.php.php-codesniffer.annotated-ruleset.php

**Comment by sajov** [comment]
i mean, 
by excluding Squiz.Commenting.FileComment.MissingSubpackageTag like above
or like you advice
```
<rule ref="Squiz.Commenting.FileComment.MissingSubpackageTag">
    <severity>0</severity>
</rule>
```

makes no change i still get an error
```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 6 | ERROR | The tag in position 2 should be the @subpackage tag
 7 | ERROR | Missing @subpackage tag in file comment
----------------------------------------------------------------------
```

**Comment by aik099** [comment]
I'm not sure what's so special about this sniff, but it does work for every sniff I previously encountered. See: https://github.com/aik099/CodingStandard/blob/master/CodingStandard/ruleset.xml#L136-L138

**Comment by aik099** [comment]
Are you certain that errors you got have `Squiz.Commenting.FileComment.MissingSubpackageTag` code? You can verify that by adding `-s` option to `phpcs` command, that will show sniff codes in reports as well (not only error message).

**Comment by sajov** [comment]
ok, thanks.
i did it wrong and used 
Squiz.Commenting.FileComment.MissingSubpackageTag 
instead of 
Squiz.Commenting.FileComment.Missing@subpackageTag

both works
```
  <rule ref="Squiz.Commenting.FileComment.Missing@subpackageTag">
    <severity>0</severity>
 </rule>
```
and 
```
  <rule ref="Squiz.Commenting.FileComment">
     <exclude name="Squiz.Commenting.FileComment.Missing@subpackageTag" />    
 </rule>
```

**Comment by gsherwood** [comment]
Those error codes shouldn't be like that because they are not consistent with the PEAR ones. I've corrected this here: https://github.com/squizlabs/PHP_CodeSniffer/commit/90ab75de01ac7ff03053b5d0bdc8cf8505ae6766

From the next PHPCS version, you'll need to use the original exclude rules that you posted and not these incorrect ones that contain `@` characters.

Thanks a lot for bringing this to my attention.

**Comment by aik099** [comment]
@gsherwood , are you sure that original exclude rules from task description are in valid format?

**Comment by gsherwood** [comment]
> are you sure that original exclude rules from task description are in valid format?

Yeah sorry. I meant to say "use the original sniff codes". So the ones without the `@` in them. Although doing this:
```xml
<exclude name="Squiz.Commenting.FileComment.MissingSubpackageTag" />
<exclude name="Squiz.Commenting.FileComment.MissingCopyrightTag" />
<exclude name="Squiz.Commenting.FileComment.IncorrectAuthor" />
<exclude name="Squiz.Commenting.FileComment.IncorrectCopyright" />
<exclude name="Squiz.Commenting.FileComment.DuplicateAuthorTag" />
```
Should also work. You can exclude a specific sniff message, which is just a shortcut for setting the severity to `0`. You can also exclude a specific sniff code from files matching a pattern. Examples on here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml

**Comment by aik099** [comment]
So `<exclude name="Squiz.Commenting.FileComment.MissingSubpackageTag" />` is equivalent to 

```xml
<rule ref="Squiz.Commenting.FileComment.MissingSubpackageTag">
    <severity>0</severity>
</rule>
```
?

**Comment by gsherwood** [comment]
That's correct. This was released in version 1.5.0.

Request is here: https://pear.php.net/bugs/bug.php?id=20091
Commit is here: https://github.com/squizlabs/PHP_CodeSniffer/commit/1d416db921cbde2f45ef06d104

**Comment by rogoit** [comment]
The -s param is great to detect the exact rule
<exclude name="Squiz.Commenting.FileComment.SubpackageTagOrder" />
